instance/instance.go

```diff
@@ -260,6 +272,32 @@ func (i *Instance) Prefix() string {
 	return i.Domain + "/"
 }
 
+// RegisterPassword replace the instance registerToken by a password
+func (i *Instance) RegisterPassword(pass string, tok string) error {
+	if i.RegisterToken == "" {
+		return errors.New("Missing registerToken")
+	}
+	if i.RegisterToken != tok {
+		return errors.New("Invalid registerToken")
+	}
+
+	hash, err := scrypt.GenerateFromPassword([]byte(pass), scrypt.DefaultParams)
+	if err != nil {
+		return err
+	}
+
+	i.RegisterToken = ""
+	i.PasswordHash = string(hash)
```

Comment:
It would be nice to have a field that says that the hash algorithm for password is scrypt. It will ease the transition to another algorithm is scrypt is found to be weak in the future. It's a recommandation from @aeris